src/query/service/src/builtin/builtin_udfs.rs

```diff
@@ -0,0 +1,169 @@
+use databend_common_ast::ast::UDFDefinition;
+use databend_common_ast::parser::parse_sql;
+use databend_common_ast::parser::tokenize_sql;
+use databend_common_ast::parser::Dialect;
+use databend_common_config::UDFConfig;
+use databend_common_exception::ErrorCode;
+use databend_common_exception::Result;
+use databend_common_expression::types::DataType;
+use databend_common_meta_app::principal::UserDefinedFunction;
+use databend_common_sql::resolve_type_name;
+use log::warn;
+
+pub struct BuiltinUDFs {
+    udf_configs: Vec<UDFConfig>,
+}
+
+impl BuiltinUDFs {
+    pub fn create(udf_configs: Vec<UDFConfig>) -> BuiltinUDFs {
+        BuiltinUDFs { udf_configs }
+    }
+
+    // Parse the UDF definition and return the UserDefinedFunction.
+    async fn parse_udf_definition(
+        &self,
+        name: &str,
+        definition: &str,
+    ) -> Result<UserDefinedFunction> {
+        let tokens = tokenize_sql(definition)?;
+        let (stmt, _) = parse_sql(&tokens, Dialect::PostgreSQL)?;
+
+        match stmt {
+            Statement::CreateUDF(ast) => match ast.definition {
+                UDFDefinition::UDFServer {
+                    arg_types,
+                    return_type,
+                    address,
+                    handler,
+                    language,
+                } => {
+                    let mut arg_datatypes = Vec::with_capacity(arg_types.len());
+                    for arg_type in arg_types {
+                        arg_datatypes.push(DataType::from(&resolve_type_name(&arg_type, true)?));
+                    }
+                    let return_type = DataType::from(&resolve_type_name(&return_type, true)?);
+                    let udf = UserDefinedFunction::create_udf_server(
+                        name,
+                        &address,
+                        &handler,
+                        &language,
+                        arg_datatypes,
+                        return_type,
+                        "Built-in UDF",
+                    );
+
+                    Ok(udf)
+                }
+                _ => Err(ErrorCode::SyntaxException(format!(
+                    "Invalid built-in UDF definition: '{}', expected UDFServer but got: {:?}",
+                    definition, ast
+                ))),
+            },
+            _ => Err(ErrorCode::SyntaxException(format!(
+                "Invalid built-in UDF definition: '{}', expected CreateUDF but got: {:?}",
+                definition, stmt
+            ))),
+        }
+    }
+
+    pub async fn to_meta_udfs(&self) -> Result<HashMap<String, UserDefinedFunction>> {
+        let mut udf_map = HashMap::new();
+
+        for udf_config in self.udf_configs.iter() {
+            match self
+                .parse_udf_definition(&udf_config.name, &udf_config.definition)
+                .await
+            {
+                Ok(user_defined_function) => {
+                    udf_map.insert(user_defined_function.name.clone(), user_defined_function);
```

Comment:
check for dup names here?

Reply:
If the built-in udfs config not invalid, the global init will log a error and skip, duplicate is no need check here.